.github/workflows/k8s-content-pr.yaml

```diff
@@ -15,12 +13,35 @@ jobs:
     outputs:
       pr-number: ${{ steps.get-pr-number.outputs.pr-number }}
     steps:
-      - name: Get PR number
-        id: get-pr-number
-        run: echo "::set-output name=pr-number::$(echo ${GITHUB_REF#refs/pull/} | cut -d'/' -f1)"
+      - name: 'Download artifacts'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr_number"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr_number.zip`, Buffer.from(download.data));
+      - name: 'Unzip artifact'
+        run: unzip pr_number.zip
+      - name: 'Read PR number'
+        run: |
+          echo "::set-output name=pr-number::$(cat pr/pr_number)"
```

Comment:
Don't we need to checkout the PR's HEAD?
Otherwise won't the container be built from the latest `master` branch?

Something like should work:
```
      - uses: actions/checkout@v4
```
According to the docs, `ref:` defaults to the reference that triggered the worflow: https://github.com/actions/checkout

Reply:
thanks, that's a good catch!